### PR TITLE
14 calculate mumax3 subregions

### PR DESCRIPTION
**Pull request by samjrholt**

### CODE REVIEW

**Review by swapneelap — COMMENTED** [review]
Left a comment. Looks good to me.

File: mumax3c/scripts/util.py [hunk 1]
```diff
@@ -1,38 +1,70 @@
...
+        region_relator[key] = []
+    unique_index = 0
+
+    for sr_index, sr_name in sr_dict.items():
+        for ms in np.unique(Ms_array[sr_indices == sr_index]):
+            if ms == 0:
+                continue
+            if unique_index > max_index:
+                raise ValueError(
+                    "mumax3 does not allow more than 256 seperate regions to be set."
```
[the remark this answers is not in the document]
Comment (lang-m) [reply mumax3c/scripts/util.py:53]: I was thinking about this as well. It does not work at the current position in the code but we could move the test after the loop to have this information and show it in the error message. Maybe also the number of unique Ms values (and subregions?).

[the remark this answers is not in the document]
Comment (lang-m) [reply mumax3c/scripts/util.py:53]: I'll move it to the end then

[the remark this answers is not in the document]
Comment (lang-m) [reply mumax3c/scripts/util.py]: Updated; please have a look

[the remark this answers is not in the document]
Comment (swapneelap) [reply mumax3c/scripts/util.py]: Looks good!

[the remark this answers is not in the document]
Comment (samjrholt) [reply mumax3c/scripts/util.py]: Sounds good Martin

**Comment by samjrholt** [comment]
@lang-m @swapneelap Please correct me if I am wrong but I thought one of the ideas was that all `ms=0` to have the same `mumax3` region? Currently we have the same output as before where `ms=0` in different subregions. I can change this but wanted to check before doing so

**Comment by samjrholt** [comment]
@lang-m @swapneelap Please correct me if I am wrong but I thought one of the ideas was that all `ms=0` to have the same `mumax3` region? Currently we have the same output as before where `ms=0` in different `subregions`. I can change this but wanted to check before doing so

**Pull request merged**